overlord/snapstate/snapstate.go

```diff
@@ -1200,6 +1200,92 @@ func UpdateMany(ctx context.Context, st *state.State, names []string, userID int
 // consider.
 type updateFilter func(*snap.Info, *SnapState) bool
 
+func rearrangeBaseKernelForSingleReboot(kernelTs, bootBaseTs *state.TaskSet) error {
+	var findTaskOfKind = func(ts *state.TaskSet, kind string) (*state.Task, int) {
```

Comment:
we could have doInstall use TaskSet.MarkEdge more and use MaybeEdge  here instead, and instead of the indexes we would mark both the link-snap and the before/after tasks